tx-pool/src/pool.rs

```diff
@@ -491,98 +346,109 @@ impl TxPool {
         let entry =
             TxEntry::new_with_timestamp(rtx, verified.cycles, verified.fee, size, timestamp);
         let tx_hash = entry.transaction().hash();
-        if self.add_proposed(entry)? {
-            Ok(CacheEntry::Completed(verified))
-        } else {
-            Err(Reject::Duplicated(tx_hash))
-        }
```

Comment:
Here we can just modify the state instead of re-adding the entry, so we can change the function signature to just pass in the id, similar to 
```
   pub(crate) fn proposed_rtx(
        &mut self.
        ...
        Id: &ProposalShortId.
    ) -> Result<CacheEntry, Reject> {
```